### DISCUSSION TITLE
Freeze successfull but signature code invalid

### DISCUSSION THREAD

**Discussion by l-lemaire** [marcelotduarte/cx_Freeze#1379] [Q&A] [answered]
Hey everyone,

I'm trying to use cx_freeze on macos.
I've been able to freeze the same application on windows no problem.

The freeze is going all well except that I get some warnings: `/Library/Developer/CommandLineTools/usr/bin/install_name_tool: warning: changes being made to the file will invalidate the code signature`

Which leads me to my error: Freeze ends successfully but when I try to run it, it crashes. No message even when launching from console but the report from the launch give me this:

```
Application Specific Information:
dyld: launch, loading dependent libraries

Dyld Error Message:
  Library not loaded: @executable_path/lib/Python
  Referenced from: /Users/USER/Documents/*/Narrator-1.0.app/Contents/MacOS/main
  Reason: no suitable image found.  Did find:
	/Users/lle/Documents/Narrator/build/Narrator-1.0.app/Contents/MacOS/lib/Python: code signature invalid for '/Users/lle/Documents/Narrator/build/Narrator-1.0.app/Contents/MacOS/lib/Python'

	/Users/lle/Documents/Narrator/build/Narrator-1.0.app/Contents/MacOS/lib/Python: stat() failed with errno=1
	/Users/lle/Documents/Narrator/build/Narrator-1.0.app/Contents/MacOS/lib/Python: code signature invalid for '/Users/lle/Documents/Narrator/build/Narrator-1.0.app/Contents/MacOS/lib/Python'

	/Users/lle/Documents/Narrator/build/Narrator-1.0.app/Contents/MacOS/lib/Python: stat() failed with errno=1

Binary Images:
       0x10144a000 -        0x10144dfff +Narrator-1.0 (0) <03713CAB-53BD-3359-B5EF-D707B5353BB2> /Users/USER/Documents/*/Narrator-1.0.app/Contents/MacOS/main
       0x1071b7000 -        0x107252fff  dyld (832.7.1) <2705F0D8-C104-3DE9-BEB5-B1EF6E28656D> /usr/lib/dyld
```

I gues it is something related to the warning I got before but I'm clueless on how to resolve this issue.

I use last version of cx-freeze (6.10 at current time) with python 3.10 on macos big sur.

Thx in advance

**Comment by marcelotduarte** [comment, marked as answer]
Using the instruction form [here](https://github.com/marcelotduarte/cx_Freeze/issues/684#issuecomment-860065222)
Could you try re-signing the app with the following command, and see if it will run then. Please let me know.

codesign --deep -f -s - /Users/lle/Documents/Narrator/build/Narrator-1.0.app

**Reply by l-lemaire** [reply]
Hey,

So running the command on the whole app didn't work as the command ended in error.
But I've been able to launch the command on the specific item that was faulty and now it is working.

Thx for your help, much appreciated.
